ui/package.json

```diff
@@ -63,13 +63,14 @@
     "why-did-you-update": "^0.1.1"
   },
   "dependencies": {
+    "@types/react": "^17.0.14",
```

Comment:
why did we add a new top level dependency?

Reply:
The reason I made some changes to some un-related packages is that I upgraded my node.js to v14 once a time because a post said some node versions have a bug causing XHR requests to never ended. With node v14, when I run `npm list|grep igv`, it will report some errors:
```
npm ERR! peer dep missing: jsoneditor@^7.1.0, required by jsoneditor-react@3.0.0
npm ERR! peer dep missing: react@^0.14.0 || ^15.0.0, required by react-rangeslider@2.2.0
npm ERR! peer dep missing: @types/react@>=16, required by react-markdown@5.0.3
├─┬ igv@2.8.6
│ ├─┬ igv-ui@1.1.7 (git+https://github.com/igvteam/igv-ui.git#b7fddfda708d2339b3bba704f8e1aeccbcd695a4)
│ │ └── igv-utils@1.2.8 (github:igvteam/igv-utils#4443005457137b0d45d77165c2fc551265522490)
│ └── igv-utils@1.2.9 (git+https://github.com/igvteam/igv-utils.git#bd1197a1733cea730875c23ff3171748cc7449bb)
```
But when I downgrade node to v11 (original version), there is no error reported.
I believe the errors do exist. So I kept the changes. For now, there are two npm errors still there:
```
npm ERR! peer dep missing: jsoneditor@^7.1.0, required by jsoneditor-react@3.0.0
npm ERR! peer dep missing: react@^0.14.0 || ^15.0.0, required by react-rangeslider@2.2.0
```
I still keep `jsoneditor@^7.1.0` because the suggested version `jsoneditor@^9.0.0` has a security flaw. Maybe I should upgrade it to  `jsoneditor@>=9.1` which the security bug is fixed. For the other error, it can't be fixed because some other packages require `react@>=16`.

Reply:
Peer dependency errors can be ignored if they aren't actually causing issues. The `jsoneditor` and `react-rangeslider` have been there forever and there is no reason to try to fix them, and unless you ACTUALLY see an error when you try to edit markdown there is no reason to try to fix the peer dependency for `react-markdown`. Please undo all the unrelated changes, as testing that there aren't unintended consequences to these version changes is really hard to do. If you are worried about a dependency vulnerability in an unrelated npm package, that change belongs in its own PR, not lumped in to a different PR where I can't properly evaluate it. But also, I do dependency audits really frequently, so in general its not something you need to worry about